internal/unionstore/art/art.go

```diff
@@ -498,7 +513,9 @@ func (t *ART) Release(h int) {
 	if h != len(t.stages) {
 		panic("cannot release staging buffer")
 	}
+	t.WriteSeqNo++
```

Comment:
Are we going to tolerance the interleaving usage between iterator and `Release`? It  won't affect the iterator I think.

Reply:
Release stage[0] can change "the snapshot" so I suppose we cannot tolerate that. WDYT?

Reply:
Agree, even the non-snapshot iterator is not affected, prohibit this case looks good to me.